Spec/PushActivationStateMachine.swift

```diff
@@ -68,19 +76,25 @@ override class var defaultTestSuite : XCTestSuite {
                 rest.internal.storage = storage
                 let stateMachine = ARTPushActivationStateMachine(rest: rest.internal, delegate: StateMachineDelegate())
                 expect(stateMachine.current).to(beAKindOf(ARTPushActivationStateAfterRegistrationSyncFailed.self))
+afterEach()
+
             }
 
             // RSH3a
-            context("State NotActivated") {
+            
+
+                func beforeEach__Activation_state_machine__State_NotActivated() {
```

Comment:
Here's an example of a `beforeEach` hook inside a context, when the scope it’s nested within _also_ has a `beforeEach` hook. You can see that we turn this hook into an appropriately-named `beforeEach__…` method (`beforeEach__Activation_state_machine__State_NotActivated` here), and at the start of _this_ method, we call the `beforeEach` of its parent scope (`beforeEach` here), if any.

And then, at the start of the test case `test__…` methods inside this context, we call the appropriate `beforeEach_…` function.